hoomd/BoxDim.h

```diff
@@ -387,6 +387,116 @@ struct
         return w;
         }
 
+    //! Compute minimum image
+    /*! \param p Position vector to compute
+        \param v Velocity vector to compute
+        \return vectors that are the minimum image vectors of \a p and \a v, obeying the periodic settings
+        \note \a p must not extend more than 1 image beyond the box
+    */
+    struct MinImage {
+        Scalar3 dx;
+        Scalar3 dv;
+    };
+
+    HOSTDEVICE MinImage minImage(const Scalar3& p, const Scalar3& v) const
```

Comment:
Rather than make a struct, put both as arguments by reference:
```suggestion
    HOSTDEVICE void minImage(Scalar3& dx, Scalar3& dv, const Scalar3& p, const Scalar3& v) const
```